lib/web3-contracts.js

```diff
@@ -131,248 +106,189 @@ export function useTokenBalance(symbol) {
       setBalance(bigNum(-1))
       return
     }
 
     cancelBalanceUpdate.current = () => {
       cancelled = true
     }
-
-    tokenContract.balanceOf(account).then(balance => {
+    const requestedAddress = address || account
+    tokenContract.balanceOf(requestedAddress).then(balance => {
       if (!cancelled) {
         setBalance(balance)
       }
     })
-  }, [account, tokenContract])
+  }, [account, address, tokenContract])
 
   useEffect(() => {
     // Always update the balance if updateBalance() has changed
     updateBalance()
 
-    if (!tokenContract || !account) {
+    if ((!account && !address) || !tokenContract) {
       return
     }
 
     const onTransfer = (from, to, value) => {
-      if (from === account || to === account) {
+      if (
+        from === account ||
+        to === account ||
+        from === address ||
+        to === address
+      ) {
         updateBalance()
       }
     }
     tokenContract.on('Transfer', onTransfer)
 
-    // Filter transfers from and to the account
-    // const filters = [
-    //   tokenContract.filters.Transfer(account),
-    //   tokenContract.filters.Transfer(null, account),
-    // ]
-    // filters.forEach(filter => tokenContract.on(filter, onTransfer))
-
     return () => {
       tokenContract.removeListener('Transfer', onTransfer)
-
-      // filters.forEach(filter =>
-      //   tokenContract.removeListener(filter, onTransfer)
-      // )
     }
-  }, [account, tokenContract, updateBalance])
+  }, [account, address, tokenContract, updateBalance])
 
   return balance
 }
 
-export function useJurorRegistryAnjBalance() {
-  const { account } = useWeb3Connect()
-  const [balance, setBalance] = useState(bigNum(-1))
+export function useBondingCurvePrice(amount, forwards = true) {
+  const [loading, setLoading] = useState(false)
+  const [price, setPrice] = useState(bigNum(-1))
 
-  const tokenContract = useKnownContract('TOKEN_ANJ')
-  const jurorsRegistryContract = useKnownContract('JURORS_REGISTRY')
-  const wrapperContract = useKnownContract('WRAPPER')
+  const bancorContract = useKnownContract('BANCOR_FORMULA')
 
-  const cancelBalanceUpdate = useRef(null)
+  const [treasuryAddress] = getKnownContract('BONDING_CURVE_TREASURY')
+  const antTreasuryBalance = useTokenBalance('ANT', treasuryAddress)
 
-  const updateBalance = useCallback(() => {
-    let cancelled = false
-
-    if (cancelBalanceUpdate.current) {
-      cancelBalanceUpdate.current()
-      cancelBalanceUpdate.current = null
-    }
-
-    if (!account || !tokenContract || !jurorsRegistryContract) {
-      setBalance(bigNum(-1))
-      return
-    }
-
-    cancelBalanceUpdate.current = () => {
-      cancelled = true
-    }
-
-    jurorsRegistryContract.balanceOf(account).then(([activeBalance]) => {
-      if (!cancelled) {
-        setBalance(activeBalance)
-      }
-    })
-  }, [account, tokenContract, jurorsRegistryContract])
+  const { loading: supplyLoading, totalSupply } = useAnjTotalSupply()
```

Comment:
Given that there's potential confusion around which token we're talking about, I'd prefix these variables with `anj`